airflow/models.py

```diff
@@ -2034,20 +2052,50 @@ def date_range(self, start_date, num=None, end_date=datetime.now()):
             start_date=start_date, end_date=end_date,
             num=num, delta=self._schedule_interval)
 
-    def following_schedule(self, dttm):
+    def directly_following_schedule(self, dttm):
         if isinstance(self._schedule_interval, six.string_types):
             cron = croniter(self._schedule_interval, dttm)
             return cron.get_next(datetime)
         elif isinstance(self._schedule_interval, timedelta):
             return dttm + self._schedule_interval
 
-    def previous_schedule(self, dttm):
+    def following_schedule(self, reference, n=1):
+        """
+        Returns the `n`th schedule after `reference`
+        :param reference: the reference schedule
+        :type reference: datetime
+        :param n: the number of schedules to go forward in time to
+        :type n: int
+        """
+        assert n >= 0, "{n} should be positive".format(**locals())
```

Comment:
actually, can't we do something like
```
self.date_range(start_date=reference, num=n+1)[-1]
``` 
or something here?

That would let us delete `directly_following_schedule`, and re-use the `date_range` "croniter" in utils